src/overlays/actors/ovl_Bg_Ydan_Hasi/z_bg_ydan_hasi.c

```diff
@@ -9,7 +9,13 @@ void BgYdanHasi_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BgYdanHasi_Update(Actor* thisx, GlobalContext* globalCtx);
 void BgYdanHasi_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
+void BgYdanHasi_initWater(BgYdanHasi* this, GlobalContext* globalCtx);
+void BgYdanHasi_updateFloatingBlock(BgYdanHasi* this, GlobalContext* globalCtx);
+void BgYdanHasi_setupThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx);
+WaterBox* BgYdanHasi_updateWater(BgYdanHasi* this, GlobalContext* globalCtx);
+void BgYdanHasi_decWaterTimer(BgYdanHasi* this, GlobalContext* globalCtx);
+void BgYdanHasi_updateThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx);
```

Comment:
Function names should follow the SystemName_FuncName naming scheme, so they should always have an uppercase letter after the underscore.

Also action names should generally be a verb describing the action itself, like "Float" or "Wait", or something more complicated if needed.